internal/minipipeline/analysis.go

```diff
@@ -555,18 +552,15 @@ func (wa *WebAnalysis) tlsComputeMetrics(c *WebObservationsContainer) {
 		}
 
 		// handle the case where both the probe and the control fail
-		if obs.TLSHandshakeFailure.Unwrap() != "" && obs.ControlTCPConnectFailure.Unwrap() != "" {
+		//
+		// See https://github.com/ooni/probe/issues/2300.
+		if obs.TLSHandshakeFailure.Unwrap() != "" && obs.ControlTLSHandshakeFailure.Unwrap() != "" {
```

Comment:
Bug `#1` is that we were checking against `ControlTCPConnectFailure`, which was wrong.